src/oas3/accessors.ts

```diff
@@ -1,13 +1,23 @@
-import { get, pick, values } from 'lodash';
+import { get } from 'lodash';
 import { OpenAPIObject, OperationObject, SecuritySchemeObject } from 'openapi3-ts';
 
-import { uniqFlatMap } from '../utils';
+import { mapToKeys } from '../utils';
 
 export function getSecurities(
   spec: Partial<OpenAPIObject>,
   operation: Partial<OperationObject>,
-): SecuritySchemeObject[] {
-  const globalSchemes = uniqFlatMap(spec.security);
-  const operationSchemes = uniqFlatMap(operation.security);
-  return values(pick(get(spec, 'components.securitySchemes'), operation.security ? operationSchemes : globalSchemes));
+): SecuritySchemeObject[][] {
+  const globalSchemes = mapToKeys(spec.security);
+  const operationSchemes = mapToKeys(operation.security);
+
+  const opSchemesPairs = operation.security ? operationSchemes : globalSchemes;
+  const definitions = get(spec, 'components.securitySchemes');
```

Comment:
You can provide a default value so you can avoid to check for `!definitions`
```suggestion
  const definitions = get(spec, 'components.securitySchemes', []);
```

Reply:
@XVincentX, this is not the same though. 

If I do this, then we have a problem here:
https://github.com/stoplightio/http-spec/blob/52e16cc15df040e227f06de324a951d19103c54f/src/oas3/__tests__/accessors.test.ts#L62
This test does not provide `securitySchemes`. And if I add that default value of `[]`, then we end up with the test resulting in:

```
- Snapshot
+ Received

- Array []
+ Array [
+   Array [
+     undefined,
+   ],
+ ]
```

This happens because the operation uses `specScheme` for security. If we provide `securitySchemes` as [], then we end up with `[]['specScheme']`.

Reply:
Aaah yeah I see what you mean. My bad. This is ok.